lib/rubocop/cop/lint/uri_regexp.rb

```diff
@@ -43,31 +45,31 @@ def on_send(node)
           end
         end
 
-        def autocorrect(node)
-          lambda do |corrector|
-            if (captured_values = uri_regexp_with_argument?(node))
-            else
-              captured_values = uri_regexp_without_argument?(node)
-            end
-
-            double_colon, arg = captured_values
+        private
 
-            top_level = double_colon ? '::' : ''
-            argument = arg ? "(#{arg.source})" : ''
+        def register_offense(node, top_level: '', arg: '')
+          format = format(MSG, top_level: top_level, arg: arg)
 
-            corrector.replace(
-              node,
-              "#{top_level}URI::DEFAULT_PARSER.make_regexp#{argument}"
-            )
+          add_offense(node.loc.selector, message: format) do |corrector|
+            autocorrect(corrector, node)
           end
         end
 
-        private
+        def autocorrect(corrector, node)
+          if (captured_values = uri_regexp_with_argument?(node))
```

Comment:
I'm surprised we don't have a cop to change this to 
```
captured_values = uri_regexp_with_argument?(node) ||
                  uri_regexp_without_argument?(node)
```

Reply:
Sure! I've also improved other redundant logic in uri_regexp.rb. Thank you for your review ❤️